logical/plugin/backend_server.go

```diff
@@ -0,0 +1,146 @@
+package plugin
+
+import (
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/logical"
+)
+
+type backendPluginServer struct {
+	broker  *plugin.MuxBroker
+	backend logical.Backend
+}
+
+func (b *backendPluginServer) HandleRequest(args *HandleRequestArgs, reply *HandleRequestReply) error {
+	conn, err := b.broker.Dial(args.StorageID)
+	if err != nil {
+		*reply = HandleRequestReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	c := rpc.NewClient(conn)
+	defer c.Close()
+
+	storage := &StorageClient{client: c}
+	args.Request.Storage = storage
+
+	resp, err := b.backend.HandleRequest(args.Request)
+	*reply = HandleRequestReply{
+		Response: resp,
+		Error:    plugin.NewBasicError(err),
+	}
+
+	return nil
+}
+
+func (b *backendPluginServer) SpecialPaths(_ interface{}, reply *SpecialPathsReply) error {
+	*reply = SpecialPathsReply{
+		Paths: b.backend.SpecialPaths(),
+	}
+	return nil
+}
+
+func (b *backendPluginServer) Logger(_ interface{}, reply *BackendLoggerReply) error {
+	*reply = BackendLoggerReply{
+		Logger: b.backend.Logger(),
+	}
+	return nil
+}
+
+func (b *backendPluginServer) HandleExistenceCheck(args *HandleExistenceCheckArgs, reply *HandleExistenceCheckReply) error {
+	conn, err := b.broker.Dial(args.StorageID)
+	if err != nil {
+		*reply = HandleExistenceCheckReply{
+			Error: plugin.NewBasicError(err),
+		}
+		return nil
+	}
+	c := rpc.NewClient(conn)
+	defer c.Close()
+
+	storage := &StorageClient{client: c}
+	args.Request.Storage = storage
+
+	checkFound, exists, err := b.backend.HandleExistenceCheck(args.Request)
+	*reply = HandleExistenceCheckReply{
+		CheckFound: checkFound,
+		Exists:     exists,
+		Error:      plugin.NewBasicError(err),
+	}
+
+	return nil
+}
+
+func (b *backendPluginServer) Cleanup(_ interface{}, _ *struct{}) error {
+	b.backend.Cleanup()
+	return nil
+}
+
+func (b *backendPluginServer) Initialize(_ interface{}, _ *struct{}) error {
+	err := b.backend.Initialize()
+	return err
+}
+
+func (b *backendPluginServer) InvalidateKey(args string, _ *struct{}) error {
+	b.backend.InvalidateKey(args)
+	return nil
+}
+
+func (b *backendPluginServer) Configure(args *ConfigureArgs, reply *ConfigureReply) error {
```

Comment:
I was thinking, if instead of the plugin’s Factory method being called here: https://github.com/hashicorp/vault/pull/2874/files#diff-91c8de17b557199c4bed14f1889e56adR17

If BackendPlugin.Server() just passed the Factory function into the `backendPluginServer` object then `backendPluginServer` could call `Factory(conf)` for the plugin, and that would remove the need for the Configure function.

And plugin backends and normal backends would, _i think_, be interoperable at that point